app/Services/RestApiClient.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Gladiator\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+
+class RestApiClient
+{
+    protected $client;
+
+    /**
+     * RestApiClient constructor.
+     *
+     * @param string  $base_uri
+     * @param array  $headers
+     */
+    public function __construct($base_uri, $headers = [])
+    {
+        $headers += [
```

Comment:
hehe :smile: adds to the array and if it's an empty array then doesn't add any items